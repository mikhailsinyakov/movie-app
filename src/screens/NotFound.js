import React, { Fragment } from "react";
import styled from "styled-components";
import Header from "components/Header";
import Settings from "components/Settings";
import NotFound from "components/NotFound";
import Footer from "components/Footer";

const Main = styled.main`
  position: relative;
  flex-grow: 1;
`;

const NotFoundPage = () => (
  <Fragment>
    <Header />
    <Main>
      <Settings />
      <NotFound />
    </Main>
    <Footer />
  </Fragment>
);

export default NotFoundPage;
