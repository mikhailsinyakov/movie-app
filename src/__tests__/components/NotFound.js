import React from "react";
import { MemoryRouter as Router } from "react-router-dom";
import Renderer from "react-test-renderer";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import withLanguage from "components/Language";
import NotFound from "components/NotFound";

const Parent = ({ children }) => <div>{children}</div>;
const ParentWithLanguage = withLanguage(Parent);

describe("Snapshot", () => {
  let component;
  beforeEach(() => {
    component = Renderer.create(
      <Router>
        <ParentWithLanguage>
          <NotFound />
        </ParentWithLanguage>
      </Router>
    );
  });
  afterEach(() => {
    component.unmount();
  });
  it("matches snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});

const renderComponent = () =>
  render(
    <Router>
      <ParentWithLanguage>
        <NotFound />
      </ParentWithLanguage>
    </Router>
  );

describe("NotFound", () => {
  it("renders page-not-found image and link", () => {
    const { getByText, getByAltText } = renderComponent();
    expect(getByAltText(/page-not-found/i)).toBeInTheDocument();
    expect(getByText(/go to the main page/i)).toBeInTheDocument();
  });
});
