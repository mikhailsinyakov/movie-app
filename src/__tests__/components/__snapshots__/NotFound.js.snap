// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot matches snapshot 1`] = `
<div>
  <div
    className="sc-AxhCb coVOru"
  >
    <div
      className="sc-AxjAm jhoanR"
    >
      <img
        alt="page-not-found"
        className="sc-AxirZ gjMfuY"
        src="img/page-not-found-en.png"
      />
    </div>
    <a
      className="sc-AxiKw cqOour"
      href="/"
      onClick={[Function]}
    >
      Go to the main page
    </a>
  </div>
</div>
`;
